src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -187,18 +190,27 @@ public async Task<JObject> Resolve(string varName, CancellationToken token)
             if (scopeCache.ObjectFields.TryGetValue(varName, out JObject valueRet))
                 return await GetValueFromObject(valueRet, token);
 
-            string[] parts = varName.Split(".");
-            if (parts.Length == 0)
-                return null;
+            ArraySegment<string> parts = new ArraySegment<string>(varName.Split(".", StringSplitOptions.TrimEntries));
+            if (parts.Count == 0 || string.IsNullOrEmpty(parts[0]))
+                throw new ReturnAsErrorException($"Failed to resolve expression: {varName}", "ReferenceError");
 
             JObject retObject = await ResolveAsLocalOrThisMember(parts[0]);
-            if (retObject != null && parts.Length > 1)
-                retObject = await ResolveAsInstanceMember(string.Join('.', parts[1..]), retObject);
+            bool hasNullCondition = parts[0][^1] == '?';
```

Comment:
I would suggest `bool throwOnNullReference = parts[0][^1] != '?';`, because `throw on null reference` is the behavior that is relevant to us.